src/crawlee/storages/_request_source_tandem.py

```diff
@@ -0,0 +1,109 @@
+from __future__ import annotations
+
+from datetime import timedelta
+from logging import getLogger
+from typing import TYPE_CHECKING
+
+from typing_extensions import override
+
+from crawlee.storages._request_provider import RequestProvider
+
+if TYPE_CHECKING:
+    from collections.abc import Sequence
+
+    from crawlee import Request
+    from crawlee.base_storage_client._models import ProcessedRequest
+    from crawlee.storages._request_source import RequestSource
+
+
+logger = getLogger(__name__)
+
+
+class RequestSourceTandem(RequestProvider):
+    """Implements a tandem behaviour for a pair of `RequestSource` and `RequestProvider`.
+
+    In this scenario, the contents of the "source" get transferred into the "provider", allowing processing the requests
+    from both sources and also enqueueing new requests (not possible with plain `RequestSource`).
+    """
+
+    def __init__(self, request_source: RequestSource, request_provider: RequestProvider) -> None:
+        self._read_only_source = request_source
+        self._read_write_provider = request_provider
+
+    @property
+    @override
+    def name(self) -> str:
+        return ''
```

Comment:
so it does not implement the interface as a whole

Reply:
Well, true. We could either remove `name` from the `RequestProvider` - it doesn't seem to be used anywhere - or we could just defer to the underlying request provider. I'm inclined towards 1.

Reply:
Agreed.